### PR TITLE
Mutate

### PR DESCRIPTION
**Pull request by SammyRobensParadise**

### CODE REVIEW

**Comment by SammyRobensParadise** [comment]
@maryrsmcphee  I think this might be close to the solution for solving the problem.. in the loop it outputs correctly but outside it doesnt?? I need to figure it out...

**Comment by maryrsmcphee** [comment]
the changes look good to me :) 

I don't totally understand why we wouldn't need a new instance of droneRecord in the `insertFront` function?

**Comment by SammyRobensParadise** [comment]
i

> the changes look good to me :)
> 
> I don't totally understand why we wouldn't need a new instance of droneRecord in the `insertFront` function?

it is being passed in

File: Lab2/lab2_drones_manager.cpp [hunk 1]
```diff
@@ -109,13 +109,12 @@ bool DronesManager::insert_front(DroneRecord value) {
...
         return true;
     } else {
-        value.next = first->next;
-        first = &value;
-        if (value.next != NULL) {
-            value.next->prev = &value;
-        }
-        first->prev = NULL;
-        last->next = NULL;
+        value.next = (first);
```
Comment (maryrsmcphee) [inline Lab2/lab2_drones_manager.cpp:112]: re creating an instance of a new droneRecord class... 
```
DroneRecord* current = first; 
DroneRecord* newDrone = new DroneRecord();
newDrone->droneID =  value.droneID;
newDrone->yearBought = value.yearBought;
newDrone->range = value.range;
		
first = newDrone;
newDrone->next = current;
newDrone->prev = NULL;
current->prev = newDrone;
``` 
Does this make sense? That way it's a bit easier to manipulate and we don't need the if statement?

**Review by maryrsmcphee — APPROVED** [review]

**Pull request merged**